src/lib/UserPositionTracker.js

```diff
@@ -0,0 +1,54 @@
+import { shouldLocate, addListener } from './savedState';
+
+/**
+ * The UserPositionTracker allows tracking changes to the user position.
+ * It respects the global setting for tracking on the map, and only
+ * collects positions while it is turned on.
+ */
+export default class UserPositionTracker {
+  userLocation: null | Position = null;
+
+  constructor() {
+    if (typeof navigator === 'undefined' || !navigator.geolocation) {
+      return;
+    }
+    addListener(this.handleStorageChange);
+    this.handleStorageChange();
+  }
+
+  handleStorageChange = () => {
+    if (shouldLocate()) {
+      this.start();
+    } else {
+      this.stop();
+    }
+  };
+
+  start() {
+    if (this._watch) {
```

Comment:
+0: Curiosity: Why the underscored variable convention?

Reply:
Not really needed, will change. Was meant to be private :)

Reply:
Commented.